docs/data/material/components/material-icons/SearchIcons.js

```diff
@@ -518,29 +518,31 @@ const searchIndex = new FlexSearchIndex({
   tokenize: 'full',
 });
 
+const nameCleanRegex = /(Outlined|TwoTone|Rounded|Sharp)$/;
+
 const allIconsMap = {};
 const allIcons = Object.keys(mui)
   .sort()
   .map((importName) => {
     let theme;
-    if (importName.includes('Outlined')) {
+    if (importName.endsWith('Outlined')) {
       theme = 'Outlined';
-    } else if (importName.includes('TwoTone')) {
+    } else if (importName.endsWith('TwoTone')) {
       theme = 'Two tone';
-    } else if (importName.includes('Rounded')) {
+    } else if (importName.endsWith('Rounded')) {
       theme = 'Rounded';
-    } else if (importName.includes('Sharp')) {
+    } else if (importName.endsWith('Sharp')) {
       theme = 'Sharp';
     } else {
       theme = 'Filled';
     }
 
-    const name = importName.replace(/(Outlined|TwoTone|Rounded|Sharp)$/, '');
+    const name = importName.replace(nameCleanRegex, '');
     let searchable = name;
     if (synonyms[searchable]) {
       searchable += ` ${synonyms[searchable]}`;
     }
-    searchIndex.addAsync(importName, searchable);
+    searchIndex.add(importName, searchable);
```

Comment:
The change that yields this outcome.

Reply:
That whole initialization block becomes about 3 times slower for me now. It's indexing everything synchronously during module evaluation. Perhaps we can defer it? Trying a few things in https://github.com/mui/material-ui/pull/44025.

Reply:
> That whole initialization block becomes about 3 times slower for me now. It's indexing everything synchronously during module evaluation.

@Janpot Ah, ok, this PR description wasn't super clear, I have update updated it with what I was measuring. My goal was to show how time to interactive went from 6.2s to 1.9s.

> It's indexing everything synchronously during module evaluation. Perhaps we can defer it as a whole? Trying a few things in https://github.com/mui/material-ui/pull/44025.

This could be even better, it's unclear to me if we need to go as far. I would be curious to merge this, wait for 30 days to see how the field vitals are and see if we need to allocate more time on this.

Reply:
> Ah, ok, this PR description wasn't super clear, I have updated it with what I was measuring. My goal was to show how time to interactive went from 6.2s to 1.9s.

It was clear, I assume it's caused by the event loop being saturated. I just meant that the module evaluation was blocked for longer. It impacts hydration but the page should be visible because SSR.

> Overall, I find it strange that we need 700ms to index 6,000 records. This feels broken. Maybe we need a different search engine too.

👍